service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalogHandler.java

```diff
@@ -754,7 +769,199 @@ public LoadTableResponse updateTable(
     if (isStaticFacade(catalog)) {
       throw new BadRequestException("Cannot update table on static-facade external catalogs.");
     }
-    return CatalogHandlers.updateTable(baseCatalog, tableIdentifier, applyUpdateFilters(request));
+    // TODO: pending discussion if table property is right way, or a writer specific knob is
+    // required.
+    return updateTableWithRollback(baseCatalog, tableIdentifier, applyUpdateFilters(request));
+  }
+
+  private static TableMetadata create(TableOperations ops, UpdateTableRequest request) {
+    request.requirements().forEach((requirement) -> requirement.validate(ops.current()));
+    Optional<Integer> formatVersion =
+        request.updates().stream()
+            .filter((update) -> update instanceof MetadataUpdate.UpgradeFormatVersion)
+            .map((update) -> ((MetadataUpdate.UpgradeFormatVersion) update).formatVersion())
+            .findFirst();
+    TableMetadata.Builder builder =
+        (TableMetadata.Builder)
+            formatVersion
+                .map(TableMetadata::buildFromEmpty)
+                .orElseGet(TableMetadata::buildFromEmpty);
+    request.updates().forEach((update) -> update.applyTo(builder));
+    ops.commit((TableMetadata) null, builder.build());
+    return ops.current();
+  }
+
+  // TODO: Clean this up when CatalogHandler become extensible.
+  // Copy of CatalogHandler#update
+  private static LoadTableResponse updateTableWithRollback(
```

Comment:
Can we just use `baseCatalog` instead of making this static?

Reply:
This is mostly taken [CatalogHandler#updateTable](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java#L342) and it calling [CatalogHandler#commit](https://github.com/apache/iceberg/blob/main/core/src/main/java/org/apache/iceberg/rest/CatalogHandlers.java#L409) which is protected, so they are exact same except the part of rollback-logic

Reply:
I know the name is a little confusing, but I don't think we really want to copy too much `CatalogHandler` logic into `IcebergCatalogHandler` -- previously, this class was named like `IcebergCatalogHandlerWrapper`. It's okay to copy small methods where we need to override behavior imo, but this is a lot of logic.

Reply:
Ack, I agree, I hope this methods were public and hence extensible.